src/sys/unix.rs

```diff
@@ -1391,6 +1391,39 @@ pub(crate) const fn to_mreqn(
     }
 }
 
+/// Possibles state for Path Maximum Transmission Unit (PMTU) Discovering of packets received on a socket.
+///
+/// It set in the IP packet Header the flag DF also known as "do not fragment".
+///
+/// For UDP it's particullary important to note that during Path MTU discovery is done by the OS,
+/// incomming datagram may be dropped.
+///
+/// ## Portability
+///
+/// It's Linux only way of managing the Path MTU.
+#[cfg(all(feature = "all", target_os = "linux"))]
+#[cfg_attr(docsrs, doc(cfg(all(feature = "all", target_os = "linux"))))]
+#[repr(C)]
+#[derive(Debug)]
+pub enum PathMtuDiscoveringMode {
```

Comment:
I'm not sure about this enum.

Maybe it's better as a type alias with constants for the options? This way we don't have to do the conversion and we don't have to deal with the case where the kernel returns a value we don't support yet.

Reply:
Ok I agree, I may do a patch to the linux man page since some are not explicitly documented outside of the `in.h`/`in6.h`

Reply:
> Maybe it's better as a type alias with constants for the options? This way we don't have to do the conversion and we don't have to deal with the case where the kernel returns a value we don't support yet.

I reused the pattern used for `Type` and `Protocol` code tell me if wasn't the best move. :)